### PR TITLE
[FMV][AArch64] Remove feature sha1 from FMV.

### PR DESCRIPTION
**Pull request by labrinea**
Sha1 has been unified with sha2 in the ACLE spec
(see https://github.com/ARM-software/acle/pull/347)
so I am changing the compiler to adhere to it.

### CODE REVIEW

File: compiler-rt/lib/builtins/cpu_model/AArch64CPUFeatures.inc [hunk 1]
```diff
@@ -33,7 +33,6 @@ enum CPUFeatures {
   FEAT_FP,
   FEAT_SIMD,
   FEAT_CRC,
-  FEAT_SHA1,
```
Comment (efriedma-quic) [inline compiler-rt/lib/builtins/cpu_model/AArch64CPUFeatures.inc:36]: Are there ABI stability concerns if you modify this enum?

Comment (labrinea) [reply compiler-rt/lib/builtins/cpu_model/AArch64CPUFeatures.inc:36]: It was raised on another pr, so perhaps it is.

Comment (tmatheson-arm) [reply compiler-rt/lib/builtins/cpu_model/AArch64CPUFeatures.inc:36]: This is an ABI break, but FMV still very much in beta status so we can make changes to the ABI. My understanding is that compiler-rt is not required to be compatible with versions of clang/llvm other than the one it ships with (please correct me if that's not the case). And the clang and gcc implementations of FMV are already ABI incompatible for other reasons.

Comment (andrewcarlotti) [reply compiler-rt/lib/builtins/cpu_model/AArch64CPUFeatures.inc:36]: > And the clang and gcc implementations of FMV are already ABI incompatible for other reasons.
Are they? I'm not currently aware of any incompatibilities.

Comment (efriedma-quic) [reply compiler-rt/lib/builtins/cpu_model/AArch64CPUFeatures.inc:36]: We've generally been more conservative with compiler-rt.builtins compared with other parts of compiler-rt, like  sanitizers.  People do ship static libraries to other people, and we don't want them to silently break.

Even if you still consider the ABI "beta", it's not clear to me how we expect users to know; is the "beta" status mentioned anywhere in LLVM/clang documentation?

**Comment by labrinea** [comment]
ping

**Review by tommat01 — APPROVED** [review]

**Comment by labrinea** [comment]
Fix here -> https://github.com/llvm/llvm-project/pull/110291

**Pull request merged**